### DISCUSSION TITLE
withSetter("assign") ?

### DISCUSSION THREAD

**Discussion by Coffee2JS** [xaviergonz/mobx-keystone#268] [Q&A] [answered]
What's the difference between

`prop<string>('').withSetter() `

and

`prop<string>('').withSetter('assign') `

?

Current doc only talks about the first one, but the 'assign' variant is still in the api... I stumbled across it when upgrading to latest version & looking at changelog.

**Comment by sisp** [comment, marked as answer]
`prop<string>('').withSetter()` automatically creates a setter method for you, e.g. for a prop named `x` you can do `model.setX(...)` without defining `setX` yourself, whereas `prop<string>('').withSetter('assign')` lets you _assign_ a new value, e.g. `model.x = ...`.

**Reply by sisp** [reply]
Would the change in https://github.com/xaviergonz/mobx-keystone/pull/269 make it clearer?
